src/Framework/test/PackageTests.cs

```diff
@@ -0,0 +1,98 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO.Compression;
+using System.Linq;
+using System.Reflection.Metadata;
+using System.Reflection.PortableExecutable;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit.Abstractions;
+
+namespace Microsoft.AspNetCore;
+
+public class PackageTests
+{
+    private readonly string _packageLayoutRoot;
+    private readonly ITestOutputHelper _output;
+
+    public PackageTests(ITestOutputHelper output)
+    {
+        _output = output;
+        var packageRoot = TestData.GetPackagesFolder();
+        _packageLayoutRoot = TestData.GetTestDataValue("PackageLayoutRoot");
+        var packages = Directory
+                        .GetFiles(packageRoot, "*.nupkg", SearchOption.AllDirectories)
+                        .Where(file => !file.EndsWith(".symbols.nupkg", StringComparison.OrdinalIgnoreCase));
+        foreach (var package in packages)
+        {
+            var outputPath = _packageLayoutRoot + Path.GetFileNameWithoutExtension(package);
+            if (!Directory.Exists(outputPath))
+            {
+                ZipFile.ExtractToDirectory(package, outputPath);
+            }
+        }
+    }
+
+    [Fact]
+    public void PackageAssembliesHaveExpectedAssemblyVersions()
+    {
+        if (!TestData.VerifyPackageAssemblyVersions())
+        {
+            // TODO - remove this, just verifying this doesn't get hit in CI
+            Assert.True(false);
+            //return;
+        }
+
+        var versionStringWithoutPrereleaseTag = TestData.GetSharedFxVersion().Split('-', 2)[0];
+        var version = Version.Parse(versionStringWithoutPrereleaseTag);
+
+        Debugger.Launch();
+        foreach (var packageDir in Directory.GetDirectories(_packageLayoutRoot))
+        {
+            // Don't test the Shared Framework or Ref pack
+            if (packageDir.Contains("Microsoft.AspNetCore.App", StringComparison.OrdinalIgnoreCase))
+            {
+                continue;
+            }
+            var packageAssembliesDir = Path.Combine(packageDir, "lib");
```

Comment:
Alright, looks like we currently have the following `tools` assemblies in packages:

- `dotnet-sql-cache` has `dotnet-sql-cache.dll`, plus the entire set of .dll's needed to run the tool
- `Microsoft.dotnet-openapi` has `dotnet-openapi.dll`, plus `Microsoft.Build.Locator.dll` in the same folder
- `Microsoft.AspNetCore.Components.WebAssembly.Server` carries a tool folder called `BlazorDebugProxy`, which has `BrowserDebugHost.dll`, `BrowserDebugProxy.dll` (note the difference between BlazorDebugProxy and BrowserDebugProxy), and a few other .dll's needed to run the tool
- `Microsoft.AspNetCore.Components.WebAssembly.Server` carries a copy of the `BlazorDebugProxy` tool, along with a `blazor-devserver.dll` and a matching .exe, plus a large set of .dll's needed to run the tool.

So, I don't think there's an easy rule to apply to all `tools` folders that would cover everything. We could special case these 4, but that's not very future-proof. Any thoughts?

Reply:
Would it work to ignore all assemblies in `tools` that are named `System...` or `Microsoft.*`❔ Allow lists are better than deny lists but this might do the trick